### ISSUE TITLE
Don't request server endpoints on user profiles

### ISSUE THREAD

**Issue by AlfredSK** [friendica/friendica#8675]
Extracted from #8663 to move it to a separate issue.

### Expected behavior
We shoudn't request server endpoints on user profiles. For example 'https://some-server.com/users/some-user-name/.well-known/host-meta'. 

### Actual behavior
I just checked my log file to see what else we are doing. :-) I grep-ed for 'host-meta'. We are apparently requesting server endpoints on user profiles. For example 'https://some-server.com/users/some-user-name/.well-known/host-meta'. All of those requests are answered with a 404 error. And in some cases we are doing it excessively.

This doesn't make any sense. Or am I wrong?

Here is a "short" part of my log. Look at the time stamps.

```
2020-05-20T20:46:54Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"c6115a","process_id":31496}
2020-05-20T20:46:55Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"aa3493","process_id":31730}
2020-05-20T20:46:56Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"c9878a","process_id":5334}
2020-05-20T20:46:57Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"3a5aa3","process_id":31614}
2020-05-20T20:46:58Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"400aa1","process_id":23400}
2020-05-20T20:46:58Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"e9f47c","process_id":31384}
2020-05-20T20:46:59Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"5bac03","process_id":31494}
2020-05-20T20:47:01Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"f4cc6e","process_id":31385}
2020-05-20T20:47:02Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"a35129","process_id":20962}
2020-05-20T20:47:03Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"6c0a10","process_id":5334}
2020-05-20T20:47:03Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"ba573f","process_id":31262}
2020-05-20T20:47:05Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"859662","process_id":5335}
2020-05-20T20:47:06Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"796d55","process_id":31383}
2020-05-20T20:47:07Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"7622ab","process_id":5243}
2020-05-20T20:47:07Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"6b7a60","process_id":9625}
2020-05-20T20:47:07Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"5a0a23","process_id":31262}
2020-05-20T20:47:09Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"ab76e4","process_id":31494}
2020-05-20T20:47:10Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"e6e841","process_id":31262}
2020-05-20T20:47:12Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"e734ff","process_id":31383}
2020-05-20T20:47:12Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"83c664","process_id":31266}
2020-05-20T20:47:13Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"b891c4","process_id":31614}
2020-05-20T20:47:15Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"4d5c2a","process_id":31615}
2020-05-20T20:47:15Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"6ea36e","process_id":31498}
2020-05-20T20:47:19Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"1754b8","process_id":31730}
2020-05-20T20:47:19Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"d8dd84","process_id":31266}
2020-05-20T20:47:20Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"5e31f9","process_id":9513}
2020-05-20T20:47:23Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"9a1b8f","process_id":16122}
2020-05-20T20:47:24Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"43ef5f","process_id":31730}
2020-05-20T20:47:24Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"64dce1","process_id":31731}
2020-05-20T20:47:24Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"c996d2","process_id":31494}
2020-05-20T20:47:25Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"4510b6","process_id":31612}
2020-05-20T20:47:29Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"1af443","process_id":31382}
2020-05-20T20:47:31Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"9c4b4f","process_id":31498}
2020-05-20T20:47:33Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"7f3f7e","process_id":31612}
2020-05-20T20:47:34Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"5d50e9","process_id":24079}
2020-05-20T20:47:35Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"dae4c7","process_id":8670}
2020-05-20T20:47:36Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"acec53","process_id":31616}
2020-05-20T20:47:38Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"64aff9","process_id":5243}
2020-05-20T20:47:41Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"3024d3","process_id":9513}
2020-05-20T20:47:41Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"a336ac","process_id":31383}
2020-05-20T20:47:44Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"81e5ad","process_id":22979}
2020-05-20T20:47:44Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"ad5ee9","process_id":31381}
2020-05-20T20:47:44Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"30a99d","process_id":31266}
2020-05-20T20:47:46Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"1d1ee6","process_id":31384}
2020-05-20T20:47:48Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"23339c","process_id":29737}
2020-05-20T20:47:49Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"1eb6dd","process_id":22979}
2020-05-20T20:47:50Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"bfccd2","process_id":23783}
2020-05-20T20:47:50Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"029d02","process_id":31265}
2020-05-20T20:47:54Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"66b7a7","process_id":31612}
2020-05-20T20:47:55Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"febb1a","process_id":31385}
2020-05-20T20:47:55Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"34be81","process_id":31497}
2020-05-20T20:47:55Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"de82dd","process_id":31731}
2020-05-20T20:47:56Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"645897","process_id":31615}
2020-05-20T20:47:57Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"98cc8f","process_id":30998}
2020-05-20T20:47:59Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"dd604d","process_id":31265}
2020-05-20T20:48:00Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"318b5b","process_id":22979}
2020-05-20T20:48:16Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"778c63","process_id":31381}
2020-05-20T20:48:20Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"ddb242","process_id":16122}
2020-05-20T20:48:21Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"c632ae","process_id":31385}
2020-05-20T20:48:22Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"d00660","process_id":31497}
2020-05-20T20:48:25Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"fdb6c1","process_id":20962}
2020-05-20T20:48:27Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"2d96b4","process_id":8670}
2020-05-20T20:48:30Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"cf7f16","process_id":31497}
2020-05-20T20:48:31Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"e36e3a","process_id":31262}
2020-05-20T20:48:32Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"2848c8","process_id":5335}
2020-05-20T20:48:32Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"ddb3ab","process_id":30998}
2020-05-20T20:48:38Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"635e1e","process_id":23783}
2020-05-20T20:48:38Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"27c753","process_id":31383}
2020-05-20T20:48:41Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"3b42af","process_id":30998}
2020-05-20T20:48:51Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"2dbef1","process_id":31499}
2020-05-20T20:48:53Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"72f3b6","process_id":31496}
2020-05-20T20:48:54Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"52e3d0","process_id":23783}
2020-05-20T20:48:54Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"f03f52","process_id":31615}
2020-05-20T20:48:57Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"491584","process_id":31617}
2020-05-20T20:49:00Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"624423","process_id":20962}
2020-05-20T20:49:00Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"98c64f","process_id":22979}
2020-05-20T20:49:01Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"3444ee","process_id":31730}
2020-05-20T20:49:02Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"866db5","process_id":31617}
2020-05-20T20:49:04Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"00e57f","process_id":31384}
2020-05-20T20:49:04Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"847f95","process_id":9513}
2020-05-20T20:49:05Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"d5b62f","process_id":31496}
2020-05-20T20:49:06Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"c384a9","process_id":31265}
2020-05-20T20:49:09Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"f9183a","process_id":31612}
2020-05-20T20:49:11Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"f77a2f","process_id":23783}
2020-05-20T20:49:13Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"630f47","process_id":31266}
2020-05-20T20:49:16Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"bcdc55","process_id":20962}
2020-05-20T20:49:20Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"12c45e","process_id":31384}
2020-05-20T20:49:21Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"034972","process_id":31730}
2020-05-20T20:49:21Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"c54b4b","process_id":31613}
2020-05-20T20:49:23Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"f461ec","process_id":5243}
2020-05-20T20:49:23Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"161361","process_id":23400}
2020-05-20T20:49:24Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"93b7cf","process_id":30998}
2020-05-20T20:49:25Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"bc150b","process_id":31262}
2020-05-20T20:49:26Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"d81947","process_id":31383}
2020-05-20T20:49:27Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"575123","process_id":31499}
2020-05-20T20:49:27Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"56e3df","process_id":31730}
2020-05-20T20:49:28Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"40ec24","process_id":31495}
2020-05-20T20:49:28Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"cc5007","process_id":31266}
2020-05-20T20:49:30Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"d4e061","process_id":7882}
2020-05-20T20:49:30Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"740d43","process_id":31615}
2020-05-20T20:49:32Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"3d176b","process_id":24079}
2020-05-20T20:49:35Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"5ef073","process_id":13454}
2020-05-20T20:49:37Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"e488af","process_id":16122}
2020-05-20T20:49:38Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"ee9182","process_id":31383}
2020-05-20T20:49:39Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"f13384","process_id":31494}
2020-05-20T20:49:39Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"69f50d","process_id":5334}
2020-05-20T20:49:40Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"ab633f","process_id":31386}
2020-05-20T20:49:40Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"a4aeea","process_id":31497}
2020-05-20T20:49:42Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"bb560c","process_id":5334}
2020-05-20T20:49:43Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"b54a58","process_id":31266}
2020-05-20T20:49:44Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"eace5b","process_id":5335}
2020-05-20T20:49:45Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"a06be6","process_id":7882}
2020-05-20T20:49:45Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"9985f6","process_id":31499}
2020-05-20T20:49:45Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"a5a764","process_id":5334}
2020-05-20T20:49:47Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"2b8163","process_id":22979}
2020-05-20T20:49:47Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"cb3638","process_id":31263}
2020-05-20T20:49:49Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"e43dad","process_id":31612}
2020-05-20T20:49:50Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"4b4043","process_id":5243}
2020-05-20T20:49:51Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"df4b2d","process_id":31615}
2020-05-20T20:49:51Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"0f6382","process_id":16122}
2020-05-20T20:49:52Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"598ef4","process_id":31384}
2020-05-20T20:49:53Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"603c38","process_id":31612}
2020-05-20T20:49:53Z index [WARNING]: error: https://mastodon.xyz/users/mrtino/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"0ccda7","process_id":9513}
```

I checked the gserver table for urls containing '/users/' and I only found 4 rows (and deleted them).
These 4 entries had other domains. So, I don't think the issue is related to the gserver table.

The example in my above comment is an extreme one. I don't see that excessive behaviour very often. But we are requesting those URLs a lot. As far as I could see only Pleroma and Mastodon nodes are involved. But this could be due to the large share of these servers in the total number of servers.

```
MariaDB [friendica]> select url from gserver where url like '%mastodon.xyz%';
+--------------------------+
| url                      |
+--------------------------+
| https://mastodon.xyz     |
| http://notmastodon.xyz   |
| http://6-28.mastodon.xyz |
+--------------------------+
3 rows in set (0.007 sec)
```

**Examples:**

```
2020-05-21T12:13:28Z index [WARNING]: error: https://skull.website/users/triz/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"7d21f7","process_id":11612}
2020-05-21T12:13:30Z index [WARNING]: error: https://mastodon.bida.im/users/Ca_Gi/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"626485","process_id":5192}
2020-05-21T12:13:38Z index [WARNING]: error: https://quey.org/users/Lunnaris/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"23f0e1","process_id":11499}
2020-05-21T12:13:41Z index [WARNING]: error: https://aspirant.de/users/imbars/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"7cb303","process_id":11764}
2020-05-21T12:13:43Z index [WARNING]: error: https://mstdn.social/users/stux/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"9d46ef","process_id":21121}
2020-05-21T12:13:46Z index [WARNING]: error: https://mstdn.social/users/IronMan/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"06a30d","process_id":4445}
2020-05-21T12:13:47Z index [WARNING]: error: https://qoto.org/users/freemo/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"209cca","process_id":11052}
2020-05-21T12:14:08Z index [WARNING]: error: https://social.tchncs.de/users/sadmin/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"acd7c3","process_id":12155}
2020-05-21T12:14:11Z index [WARNING]: error: https://qoto.org/users/freemo/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"43964d","process_id":4445}
2020-05-21T12:14:12Z index [WARNING]: error: https://social.bau-ha.us/users/bytespeicher/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"40922d","process_id":11500}
2020-05-21T12:14:16Z index [WARNING]: error: https://octodon.social/users/TQ/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"054159","process_id":19002}
2020-05-21T12:14:26Z index [WARNING]: error: https://hackers.town/users/feonixrift/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"80832d","process_id":11765}
2020-05-21T12:14:30Z index [WARNING]: error: https://norden.social/users/Firehorse/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"98552b","process_id":12155}
2020-05-21T12:14:37Z index [WARNING]: error: https://mamot.fr/users/fipdirect/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"4b55e3","process_id":10861}
2020-05-21T12:14:39Z index [WARNING]: error: https://quey.org/users/Lunnaris/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"0a0989","process_id":11766}
2020-05-21T12:14:40Z index [WARNING]: error: https://is.a.qute.dog/users/KS/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"65ccf3","process_id":11498}
2020-05-21T12:14:45Z index [WARNING]: error: https://embassy.social/users/thalon/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"4e0801","process_id":11764}
2020-05-21T12:14:45Z index [WARNING]: error: https://norden.social/users/Firehorse/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"7a2dc7","process_id":4518}
2020-05-21T12:14:45Z index [WARNING]: error: https://koyu.space/users/koyu/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"d70b05","process_id":4596}
2020-05-21T12:14:45Z index [WARNING]: error: https://mainburg.hallertau.social/users/badde/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"b67f80","process_id":11611}
2020-05-21T12:14:45Z index [WARNING]: error: https://hackers.town/users/feonixrift/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"5ff825","process_id":11498}
2020-05-21T12:14:45Z index [WARNING]: error: https://mastodon.host/users/federationbot/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"e2d74d","process_id":4519}
2020-05-21T12:14:45Z index [WARNING]: error: https://toot.koeln/users/troetschubser/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"b8616d","process_id":11612}
2020-05-21T12:14:47Z index [WARNING]: error: https://redliberal.com/users/Josecik/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"7589c3","process_id":10861}
2020-05-21T12:14:47Z index [WARNING]: error: https://society.oftrolls.com/users/damuc2/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"f5b5db","process_id":11848}
```

**The worker is also checking these URLs for 'UpdateContact':**

```
2020-05-21T13:09:03Z index [WARNING]: error: https://mastodon.social/users/Gargron/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"37e77e","process_id":28544}
2020-05-21T13:11:09Z index [WARNING]: error: https://mastodon.social/users/Gargron/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"d464b1","process_id":28441}
2020-05-21T13:13:28Z worker [WARNING]: error: http://shiroganedon.net/users/.well-known/host-meta: 404 -  {"worker_id":"b9e88d2","worker_cmd":"UpdateContact"} - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"e93a78","process_id":10739}
2020-05-21T13:13:32Z worker [WARNING]: error: http://shiroganedon.net/users/.well-known/host-meta: 404 -  {"worker_id":"f06cd10","worker_cmd":"UpdateContact"} - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"24fc7e","process_id":10814}
2020-05-21T13:13:38Z index [WARNING]: error: https://mastodon.social/users/Gargron/.well-known/host-meta: 404 -  [] - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"51eda1","process_id":5079}
2020-05-21T13:13:40Z worker [WARNING]: error: http://shiroganedon.net/users/.well-known/host-meta: 404 -  {"worker_id":"f06cd10","worker_cmd":"UpdateContact"} - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"24fc7e","process_id":10814}
2020-05-21T13:13:41Z worker [WARNING]: error: http://shiroganedon.net/users/.well-known/host-meta: 404 -  {"worker_id":"81035fe","worker_cmd":"UpdateContact"} - {"file":"CurlResult.php","line":169,"function":"checkSuccess","uid":"c9567b","process_id":10806}
```

### Friendica version you encountered the problem
2020.06-dev

### Friendica source (git, zip)
git

### PHP version
7.3

### SQL version
10.3 MariaDB

**Comment by MrPetovan** [comment]
Well, it sorta makes sense. When we're probing a profile URL, the first thing we do is check for `/.well-known/host-meta`, on the profile URL. If it fails, then we extract the host (and eventual port number) and we try again. So we do expect the first try to fail.

Why can't we just take the host (and eventual port number)  first, you ask? Because Friendica allows subfolder installs, which means that the base domain (and eventual port number) could very well not have a host-meta, but the subdirectory containing the Friendica install would.

So we have to check every single path level of the URI we probe for a host-meta, which results in many failed requests.

The ideal would be to probe a single URL once like this, and if we found a host-meta, to not do it anymore ever again. I believe this is the point of @annando's work with `contact.gsid` where we formally attribute a profile identified by a URL to a server identified by a base URL, and once the association is done, we wouldn't have to perform that kind of recursive nightmare anymore on said profile.

Am I wrong?

**Comment by annando** [comment]
With AP contacts we could benefit from the new field, yes.

**Comment by MrPetovan** [comment]
Bottom line is we will continue to probe these non-existing URLs.

**Comment by annando** [comment]
Not for AP contacts.

**Comment by MrPetovan** [comment]
Is it implemented yet?

**Comment by annando** [comment]
I need some more minutes for it :-)
